ecommerce/views.py

```diff
@@ -187,9 +195,89 @@ def get_object(self):
         return basket
 
 
-class CouponView(APIView):
+class BulkEnrollCouponListView(APIView):
+    """
+    Admin view for fetching coupons that can be used for bulk enrollment
+    """
+
+    permission_classes = (IsAdminUser,)
+    authentication_classes = (SessionAuthentication,)
+
+    def get(self, request, *args, **kwargs):  # pylint: disable=missing-docstring
+        serialized = [
+            {
+                "coupon_payment_id": coupon_payment.pk,
+                "name": coupon_payment.name,
+                "num_coupon_codes": coupon_payment.versions.first().num_coupon_codes,
+                "product_type": product.content_type.model,
+                "title": product.content_object.title,
+                "text_id": None,
+            }
+            for coupon_payment, product in get_full_price_coupon_product_set()
+        ]
+
+        return Response(status=status.HTTP_200_OK, data=serialized)
+
+
+class BulkEnrollmentSubmitView(APIView):
+    """
+    Admin view for submitting bulk enrollment requests
+    """
+
+    permission_classes = (IsAdminUser,)
+    authentication_classes = (SessionAuthentication,)
+
+    def post(self, request, *args, **kwargs):
+        """View to send an enrollment email to all users in an uploaded csv file"""
+        coupon_payment_id = int(request.data["coupon_payment_id"])
+        coupon_payment = CouponPayment.objects.get(pk=coupon_payment_id)
+
+        # Extract emails of users to enroll from the uploaded user csv file
+        users_file = request.data["users_file"]
+        reader = csv.reader(users_file.read().decode("utf-8").splitlines())
+        emails = [email_row[0] for email_row in reader]
+
+        if len(emails) > coupon_payment.latest_version.num_coupon_codes:
+            return Response(
+                status=status.HTTP_400_BAD_REQUEST,
+                data={
+                    "errors": [
```

Comment:
This error is correctly returned when the csv file has more emails than there are coupons, but is not displayed on the page.

Reply:
Yep, forgot about this one. Will fix